src/ui/hooks.js

```diff
@@ -2,9 +2,16 @@ import deepEqual from 'fast-deep-equal';
 import { delay } from '../utils/time-utils.js';
 import { useUser } from '../components/user-context.js';
 import { hasUserPermission } from '../domain/permissions.js';
+import { useRequest } from '../components/request-context.js';
 import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { getCurrentUrlFromLocation, isBrowser } from './browser-helper.js';
 import { LOADING_SPINNER_MINIMUM_PERIOD_IN_MILLISECONDS } from '../domain/constants.js';
 
+export function useGetCurrentUrl() {
```

Comment:
This could be `useCurrentUrl`, doesn't need to be a method (just like the `useRequest` hook), right?

Reply:
Would be nice, but then we would have to watch for changes with a `useEffect` (as we do change the URL in the browser without reloading).

Reply:
But we never change the URL route, just query params probably, right?

Reply:
Oh, got it, client side navigation, which I forgot about 🤦‍♀️ Never mind, sorry for the irrelevant comment.